mlir/lib/Conversion/MemRefToSPIRV/MemRefToSPIRV.cpp

```diff
@@ -481,6 +481,14 @@ calculateRequiredAlignment(Value accessedPtr, Operation *memrefAccessOp) {
   assert((isa<memref::LoadOp, memref::StoreOp>(memrefAccessOp)) &&
          "Bad op type");
 
+  auto nontemporalAttr = memrefAccessOp->getAttrOfType<BoolAttr>("nontemporal");
```

Comment:
`nontemproal` is an attribute codified in the memref load/store op's definition. You can have direct accessors like `loadOp.getNontemproal()`.

Reply:
As this function takes `Operation *` to allow both load and store, this seems to be the most convenient way to access the attribute. I can extract `"nontemporal"` string as a constant if you can suggest a good place for it. Or would you prefer casting to `memref::LoadOp`/`memref::StoreOp` instead?

Reply:
We can make this a templated function so that we can use the accessor directly.